### DISCUSSION TITLE
Variable DirectoryStructure only works if  variable Directory provided; DatabaseBackup

### DISCUSSION THREAD

**Discussion by chris904apps** [olahallengren/sql-server-maintenance-solution#485] [Questions & Answers]
Is there a reason why `@DirectoryStructure` only works if `@Directory` provided? I want to use the directory that is configured in Sql Server but append the DirectoryStruture. Seems like a simple issue to fix but didn't know if there was a good reason why it shouldn't.

Directory Structure 
Database Backup

Code in question is 2848
![Code Section](https://user-images.githubusercontent.com/19228678/109044655-a7faab80-76a0-11eb-924a-76dd03cf4186.png)

In last IF statement The CurrentDirectoryStructure is being overwritten if Directory variable is not supplied. This doesn't seem to be correct to me. But again, There could be a case that I'm not seeing..

**Comment by Directory** [comment]
Please stop pinging me i have no idea what this repo is this is just my username leave my one j have hundreds of notifications from this repos issue tracker putting an @ infront of something conveniently called directory

**Reply by chris904apps** [reply]
Take a breath...
I apologize for the problems this may have cause you. I was unaware that you would be notified about post. I have edited the comment and incased the tsql variables that I was referencing into a code block. I will also update the title if possible.

Living and Learning, Chris
